GraknClient.java

```diff
@@ -19,63 +19,159 @@
 
 package grakn.client;
 
-import grakn.client.Grakn.Client;
-import grakn.client.Grakn.DatabaseManager;
-import grakn.client.Grakn.Session;
+import grakn.client.common.exception.GraknClientException;
+import grakn.client.rpc.Address;
 import grakn.client.rpc.RPCDatabaseManager;
 import grakn.client.rpc.RPCSession;
+import grakn.common.collection.Pair;
+import grakn.protocol.cluster.ClusterGrpc;
+import grakn.protocol.cluster.DiscoveryProto;
 import io.grpc.Channel;
 import io.grpc.ManagedChannel;
 import io.grpc.ManagedChannelBuilder;
+import io.grpc.StatusRuntimeException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import java.util.Set;
+import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
 
-public class GraknClient implements Client {
+import static grakn.client.common.exception.ErrorMessage.Client.CLUSTER_NOT_AVAILABLE;
+import static grakn.common.collection.Collections.pair;
 
-    public static final String DEFAULT_URI = "localhost:1729";
+public class GraknClient {
+    public static final String DEFAULT_ADDRESS = "localhost:1729";
 
-    private final ManagedChannel channel;
-    private final DatabaseManager databases;
+    public static class Core implements Grakn.Client {
+        private final ManagedChannel channel;
+        private final RPCDatabaseManager.Core databases;
 
-    public GraknClient() {
-        this(DEFAULT_URI);
-    }
+        public Core() {
+            this(DEFAULT_ADDRESS);
+        }
 
-    public GraknClient(String address) {
-        channel = ManagedChannelBuilder.forTarget(address).usePlaintext().build();
-        databases = new RPCDatabaseManager(channel);
-    }
+        public Core(String address) {
```

Comment:
Make the constructors private and expose static method for creating core clients:

```
GraknClient.Core client = GraknClient.core(address);
```

Reply:
See https://github.com/graknlabs/client-java/pull/232#discussion_r560924942